src/cucumberForgeDesktop.js

```diff
@@ -32,16 +32,37 @@ const getFeatureFiles = (directoryName) => {
   return featureFiles;
 };
 
+const toggleLoadingInd = () => {
+  const loadingElem = document.getElementById('loadingInd');
+  const outputElement = document.getElementById('output');
+  if (loadingElem.style.display === 'block') {
+    loadingElem.style.display = 'none';
+    outputElement.style.display = 'block';
+    document.getElementById('appBody').classList.remove('empty');
+  } else {
+    loadingElem.style.display = 'block';
+    outputElement.style.display = 'none';
+    document.getElementById('appBody').classList.add('empty');
+  }
+};
+
 const createReport = () => {
+  const startTime = Date.now();
+  toggleLoadingInd();
+
   tag = document.getElementById('tagBox').value;
   const featureFiles = getFeatureFiles(selectedFolderPath);
   generator.generate(featureFiles, projectName, tag).then((result) => {
     reportHTML = result;
+    // Display the loading indicator for at least 0.5 sec
+    const timeout = 500 - (Date.now() - startTime);
+    setTimeout(() => {
```

Comment:
could probably single-line this, and you don't need to wrap `toggleLoadingInd`, just use it as the param directly....

```js
setTimeout(toggleLoadingInd, timeout);
```

Reply:
Good call!